### PR TITLE
add shared step to create bastion ignition file

### PR DESCRIPTION
**Pull request by jianli-wei**
1. add step "ignition-bastionhost", which creates ignition file for launching bastion host (part of [JIRA ticket](https://issues.redhat.com/browse/OCPQE-9985))
2. update step "gcp-provision-bastionhost" accordingly
3. update chains accordingly, including "cucushift-installer-rehearse-gcp-ipi-private-provision", "cucushift-installer-rehearse-gcp-ipi-disconnected-provision", "cucushift-installer-rehearse-gcp-ipi-disconnected-proxy-provision"

### CODE REVIEW

**Comment by jianli-wei** [comment]
Analyzed the above failure, it's after successful installation. So nothing about the changes of the PR. 

Test name | Installation Result / Failure
-- | --
ci/rehearse/periodic-ci-openshift-verification-tests-master-nightly-4.11-e2e-gcp-ipi-private | Step phase pre succeeded after 41m51s.

File: ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+      "tls": {}
+    },
+    "timeouts": {},
+    "version": "3.0.0"
+  },
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "sshAuthorizedKeys": [
```
Comment (jianlinliu) [inline ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh:56]: I think maybe we can use the original logic. set `"sshAuthorizedKeys": []`, then update it per `ssh_pub_key`, this wold help get some benefits when `ssh_pub_key` have multiple-lines ssh keys there.
```
# update ssh keys
tmp_keys_json=`mktemp`
tmp_file=`mktemp`
echo '[]' > "$tmp_keys_json"

readarray -t contents < "${ssh_pub_keys_file}"
for ssh_key_content in "${contents[@]}"; do
  jq --arg k "$ssh_key_content" '. += [$k]' < "${tmp_keys_json}" > "${tmp_file}"
  mv "${tmp_file}" "${tmp_keys_json}"
done

jq --argjson k "`jq '.| unique' "${tmp_keys_json}"`" '.passwd.users[0].sshAuthorizedKeys = $k' < "${bastion_ignition_file}" > "${tmp_file}"
mv "${tmp_file}" "${bastion_ignition_file}"
```

**Comment by jianli-wei** [comment]
Analyzed the above failure, it's after successful installation. So nothing about the changes of the PR.

Test name | Installation Result / Failure
-- | --
ci/rehearse/periodic-ci-openshift-verification-tests-master-nightly-4.11-e2e-gcp-ipi-private | Step phase pre succeeded after 36m21s.

File: ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+echo '[]' > "$tmp_keys_json"
+
+ssh_pub_keys_file="${CLUSTER_PROFILE_DIR}/ssh-publickey"
+readarray -t contents < "${ssh_pub_keys_file}"
+for ssh_key_content in "${contents[@]}"; do
+  jq --arg k "$ssh_key_content" '. += [$k]' < "${tmp_keys_json}" > "${tmp_file}"
+  mv "${tmp_file}" "${tmp_keys_json}"
+done
+
+jq --argjson k "`jq '.| unique' "${tmp_keys_json}"`" '.passwd.users[0].sshAuthorizedKeys = $k' < "${bastion_ignition_file}" > "${tmp_file}"
```
Comment (yunjiang29) [inline ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh:388]: a minor comment,  I think one ssh key is enough for Prow CI, like https://github.com/openshift/release/blob/master/ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh#L272-L273

we do support multi-keys in Flexy, it's b/c user have such requirements, e.g. case debugging.

Comment (jianli-wei) [reply ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh:388]: @jianlinliu WDTY

Comment (jianlinliu) [reply ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh:388]: We can make the code having the capacity, that is not bad things. I am not sure if someday we also had such similar requirement. So I prefer to keep it.

**Pull request merged**